src/nock/matcher.ts

```diff
@@ -0,0 +1,268 @@
+import Ajv from 'ajv';
+import createDebug from 'debug';
+import { createSchema } from 'genson-js/dist';
+import { inspect } from 'util';
+
+import { RecordingDefinitions } from '../superface-test.interfaces';
+import { decodeResponse, getHeaderValue } from './matcher.utils';
+
+interface MatchHeaders {
+  old?: string;
+  new?: string;
+}
+
+interface ResponseHeaderMatch {
+  valid: boolean;
+  contentEncoding?: MatchHeaders;
+  contentType?: MatchHeaders;
+}
+
+const schemaValidator = new Ajv();
+const debugMatching = createDebug('superface:testing:matching');
+const debugMatchingSensitive = createDebug(
+  'superface:testing:matching:sensitive'
+);
+debugMatchingSensitive(
+  `
+WARNING: YOU HAVE ALLOWED LOGGING SENSITIVE INFORMATION.
+THIS LOGGING LEVEL DOES NOT PREVENT LEAKING SECRETS AND SHOULD NOT BE USED IF THE LOGS ARE GOING TO BE SHARED.
+CONSIDER DISABLING SENSITIVE INFORMATION LOGGING BY APPENDING THE DEBUG ENVIRONMENT VARIABLE WITH ",-*:sensitive".
+`
+);
+
+/**
+ * Matches old recording file to new recorded traffic.
+ * Assumes we always have correct order of HTTP calls.
+ *
+ * @param oldTrafficDefs - old traffic loaded from recording file
+ * @param newTrafficDefs - new recorded traffic
+ * @returns boolean representing whether recordings match or not
+ */
+export async function matchTraffic(
+  oldTrafficDefs: RecordingDefinitions,
+  newTrafficDefs: RecordingDefinitions
+): Promise<boolean> {
+  if (oldTrafficDefs.length != newTrafficDefs.length) {
+    debugMatching('Number of recorded HTTP calls do not match');
+
+    return false;
+  }
+
+  for (let i = 0; i < oldTrafficDefs.length; i++) {
+    // access old recording and new recording
+    const oldTraffic = oldTrafficDefs[i];
+    const newTraffic = newTrafficDefs[i];
+
+    debugMatching(
+      `Matching HTTP calls  ${oldTraffic.scope + oldTraffic.path} : ${
+        newTraffic.scope + newTraffic.path
+      }`
+    );
+
+    // method
+    debugMatching('\trequest method');
+    if (oldTraffic.method !== newTraffic.method) {
+      debugMatching(
+        `Request method does not match: "${
+          oldTraffic.method ?? 'not-existing'
+        }" - "${newTraffic.method ?? 'not-existing'}"`
+      );
+
+      return false;
+    }
+
+    // status
+    debugMatching('\tresponse status');
+    if (oldTraffic.status !== newTraffic.status) {
+      debugMatching(
+        `Status codes do not match: "${
+          oldTraffic.status ?? 'not-existing'
+        }" - "${newTraffic.status ?? 'not-existing'}"`
+      );
+
+      return false;
+    }
+
+    // scope
+    debugMatching('\trequest scope');
+    if (oldTraffic.scope !== newTraffic.scope) {
+      debugMatchingSensitive(
+        `Scopes do not match: "${oldTraffic.scope}" - "${newTraffic.scope}"`
+      );
+
+      return false;
+    }
```

Comment:
Not sure what you mean by leaking.

Reply:
You assume that user of testing library knows what scope means. But thats term from nock and I don't think it will be clear when someone see it in logs.

Reply:
True, I renamed it to `baseUrl` and will look out for this.